Author: PiRK <pierrekn@gmail.com>
Date:   Mon Apr 17 10:29:02 2023 +0200

    This patch fixes a compile error encountered on newer Ubuntu when building
    Qt 5.12.10.  The issue was observed on Ubuntu 21.10 and 22.04,
    where a missing include caused compile errors.

    This patch just fixes the error by including <limits>.

diff --git a/qtbase/src/corelib/global/qendian.h b/qtbase/src/corelib/global/qendian.h
--- a/qtbase/src/corelib/global/qendian.h
+++ b/qtbase/src/corelib/global/qendian.h
@@ -48,5 +48,6 @@
 #include <stdlib.h>
 #include <string.h>
+#include <limits>
 
 #ifdef min // MSVC
 #undef min

diff --git a/qtbase/src/tools/moc/generator.cpp b/qtbase/src/tools/moc/generator.cpp
--- a/qtbase/src/tools/moc/generator.cpp
+++ b/qtbase/src/tools/moc/generator.cpp
@@ -41,6 +41,7 @@
 #include <QtCore/qstringview.h>
 
 #include <math.h>
 #include <stdio.h>
+#include <limits>
 
 #include <private/qmetaobject_p.h> //for the flags.

diff -urN a/qtbase/src/corelib/tools/qbytearraymatcher.h b/qtbase/src/corelib/tools/qbytearraymatcher.h
--- a/qtbase/src/corelib/tools/qbytearraymatcher.h	2018-10-16 08:37:05.000000000 +0000
+++ b/qtbase/src/corelib/tools/qbytearraymatcher.h	2022-04-12 00:57:54.919426527 +0000
@@ -41,6 +41,7 @@
 #define QBYTEARRAYMATCHER_H
 
 #include <QtCore/qbytearray.h>
+#include <limits>
 
 QT_BEGIN_NAMESPACE
 
